src/ansys/fluent/core/streaming_services/events_streaming.py

```diff
@@ -1,30 +1,58 @@
 """Module for events management."""
 
+from enum import Enum
 from functools import partial
 import logging
-from typing import Callable, List
+from typing import Callable
 
 from ansys.api.fluent.v0 import events_pb2 as EventsProtoModule
-from ansys.fluent.core.exceptions import DisallowedValuesError, InvalidArgument
+from ansys.fluent.core.exceptions import InvalidArgument
 from ansys.fluent.core.streaming_services.streaming import StreamingService
 
 network_logger = logging.getLogger("pyfluent.networking")
 
 
-class EventsManager(StreamingService):
-    """Manages server-side events.
+class Event(Enum):
+    """Enumerates over supported server (Fluent) events."""
+
+    TIMESTEP_STARTED = "TimestepStartedEvent"
+    TIMESTEP_ENDED = "TimestepEndedEvent"
+    ITERATION_STARTED = "IterationStartedEvent"
+    ITERATION_ENDED = "IterationEndedEvent"
+    CALCULATIONS_STARTED = "CalculationsStartedEvent"
+    CALCULATIONS_ENDED = "CalculationsEndedEvent"
+    CALCULATIONS_PAUSED = "CalculationsPausedEvent"
+    CALCULATIONS_RESUMED = "CalculationsResumedEvent"
+    ABOUT_TO_READ_CASE = "AboutToReadCaseEvent"
+    CASE_READ = "CaseReadEvent"
+    ABOUT_TO_READ_DATA = "AboutToReadDataEvent"
+    DATA_READ = "DataReadEvent"
+    ABOUT_TO_INITIALIZE = "AboutToInitializeEvent"
+    INITIALIZED = "InitializedEvent"
+    REPORT_DEFINITION_CHANGED = "ReportDefinitionChangedEvent"
+    PLOT_SET_CHANGED = "PlotSetChangedEvent"
+    RESIDUAL_PLOT_CHANGED = "ResidualPlotChangedEvent"
+    CLEAR_SETTINGS_DONE = "ClearSettingsDoneEvent"
+    AUTO_PAUSE = "AutoPauseEvent"
+    PROGRESS = "ProgressEvent"
+    SOLVER_TIME_ESTIMATE = "SolverTimeEstimateEvent"
+    ERROR = "ErrorEvent"
+    COMMAND_COMPLETED = "CommandCompletedEvent"
+    DATA_MODEL_CHANGED = "DataModelChangedEvent"
+    EXECUTE_IN_CLIENT = "ExecuteInClientEvent"
```

Comment:
This is a specific solution for the _remote client_ architecture where the server is directing the client to execute arbitrary Scheme code locally. We will not have this use case so we can leave this event out at this level.